proto/message_contents/message.proto

```diff
@@ -67,4 +67,38 @@ message DecodedMessage {
     string content_topic = 6;
     ConversationReference conversation = 7;
     bytes content_bytes = 8;  // encapsulates EncodedContent
+}
+
+// Plaintext header included with messages, visible to all
+message PadlockMessageHeader {
+    uint64 sent_ns = 1;
+    // The session_id is derived via SHA256. Any other identifier will leak the
+    // sender of the message.
+    string session_id = 2;
+}
+
+// Encrypted body included with messages, only visible to recipients
+// - For safety, the sender_account_id and sender_installation_id MUST be
+//   derived from the session_id in the header by each installation receiving
+//   the message
+// - The recipient is unnecessary to include here - the recipient at the session
+//   level is the receiving installation, and the recipient at the conversation
```

Comment:
"the recipient at the conversation level is whoever is the conversation refered to by the convo_id" 
Is this true?
I would have assumed that the recipient is the account(s) which is associated by the recieving installation.

Reply:
That makes sense. I think I'm getting pedantic around conversation membership. Where the recipient is unknown with the exception that if It was sent to you it was intended for you. Any conversationId would simply hint at a possible set of recipients, rather than being a guaranteed.

I don't think it particularly matters in this case, but would have more of an impact in the context of threat and privacy modelling